src/toPairs.ts

```diff
@@ -16,22 +14,14 @@ export function toPairs<T>(object: Record<string, T>): Array<[string, T]> {
   return Object.entries(object);
 }
 
-// Inspired and largely copied from `sindresorhus/ts-extras`:
-// @see https://github.com/sindresorhus/ts-extras/blob/44f57392c5f027268330771996c4fdf9260b22d6/source/object-entries.ts
-type ObjectValues<T extends Record<PropertyKey, unknown>> =
-  Required<T>[ObjectKeys<T>];
-type ObjectEntry<T extends Record<PropertyKey, unknown>> = [
-  ObjectKeys<T>,
-  ObjectValues<T>,
-];
-type ObjectEntries<T extends Record<PropertyKey, unknown>> = Array<
-  ObjectEntry<T>
+type Pairs<T> = Array<
+  { [K in keyof T]-?: [key: K, value: Required<T>[K]] }[keyof T]
 >;
 
 export namespace toPairs {
-  export function strict<T extends Record<PropertyKey, unknown>>(
+  export function strict<const T extends Record<PropertyKey, unknown>>(
```

Comment:
With the new Pairs implementation, we don't need the narrowing to only `Record<PropertyKey, unknown>>`, it would work on any T. Also, the `const` part would not be supported in previous typescript versions we still want to support. Is it required to make the typing work?
```suggestion
  export function strict<T>(
```

Reply:
I'll check on that

Reply:
Removing the `const` part means that I have to update a test adding `as const` to the object, I dont think is too much of an issue 🤔 

Do we have any part of the docs saying the minimun ts version supported?

Reply:
Not the docs because we never made the decision public, but due to existing code and our desire to support as many use cases as possible, our effective maximum typescript version is 4.4 for any public facing api (although internal code can use the latest typescript as long as that isn't exported). We have tests to validate this is the case and we don't ship unsupported code. 

Check out our ci github action for more context.

Reply:
Removing the narrowing we have a new ts error to register to ts-expect-error (TS2769) because the original `Object.entries` expect a Record:

![image](https://github.com/remeda/remeda/assets/1144075/542d1c46-f015-431d-9899-c94ab8b5331d)

Do you think it's ok?

Reply:
It's just because of the nullabe/undefined cases, use:
```ts
  export function strict<T extends NonNullable<unknown>>(obj: T): Pairs<T> {
```